bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/v18/PsbtRpcTest.scala

```diff
@@ -0,0 +1,114 @@
+      resultF.map { result =>
+        val inputs = result.inputs
+        logger.info(inputs.toString)
+        assert(inputs.nonEmpty)
+      }
+    }
+  }
+  it should "have things in it" in {
+
+    val startF = client.start()
+    val psbt =
+      "cHNidP8BACoCAAAAAAFAQg8AAAAAABepFG6Rty1Vk+fUOR4v9E6R6YXDFkHwhwAAAAAAAA=="
+    val analyzedF = startF.flatMap(_ => client.analyzePsbt(psbt))
+
+    analyzedF.map { result =>
+      assert(result.inputs.exists(_.next.nonEmpty))
+      assert(result.inputs.exists(_.missing.head.pubkeys.head.nonEmpty))
+      assert(result.inputs.exists(_.missing.head.signatures.head.nonEmpty))
+      assert(result.inputs.exists(_.missing.head.redeemscript.nonEmpty))
+      assert(result.inputs.exists(_.missing.head.witnessscript.nonEmpty))
+      assert(result.inputs.exists(_.is_final))
+      assert(result.inputs.exists(_.has_utxo))
+      assert(result.estimated_feerate.nonEmpty)
+      assert(result.estimated_vsize.nonEmpty)
+      assert(result.fee.nonEmpty)
+      assert(result.next.nonEmpty)
+    }
+  }
+  it should "check the results of analyzepsbt " in {
+    val startF = client.start()
+    val psbt =
+      "cHNidP8BACoCAAAAAAFAQg8AAAAAABepFG6Rty1Vk+fUOR4v9E6R6YXDFkHwhwAAAAAAAA=="
+    val analyzedF = startF.flatMap(_ => client.analyzePsbt(psbt))
+    val expectedfee = Bitcoins(0.00001)
+    val expectedfeerate = 0.0000005
+    val expectedestimatedvsize = 10
+    val expectedhasutxo = true
+    val expectedisfinal = true
+    val expectedrole = "updater"
+    analyzedF.map { result =>
+      assert(result.fee.get == expectedfee)
+      assert(result.next == expectedrole)
+      assert(result.inputs.head.has_utxo == expectedhasutxo)
+
+      assert(result.inputs.head.is_final == expectedisfinal)
+
+      assert(result.estimated_vsize.get == expectedestimatedvsize)
+
+      assert(result.estimated_feerate.get == expectedfeerate)
+
+      assert(result.inputs.head.next.get == expectedrole)
+
+    }
+  }
+
+  //Todo figure out how to implement a test here
+  it should "utxoUpdateResult contains the input" in {
+    val startF = client.start()
+    val psbt =
+      "cHNidP8BACoCAAAAAAFAQg8AAAAAABepFG6Rty1Vk+fUOR4v9E6R6YXDFkHwhwAAAAAAAA=="
+    val updatedF = startF.flatMap(_ => client.utxoUpdatePsbt(psbt))
+
+    updatedF.map { result =>
+      assert(result.contains(psbt))
+    }
+  }
+
+  it should "joinpsbts " in {
+    val startF = client.start()
+    val psbt =
+      "cHNidP8BACoCAAAAAAFAQg8AAAAAABepFG6Rty1Vk+fUOR4v9E6R6YXDFkHwhwAAAAAAAA=="
+    val seqofpsbts = Seq(
+      "cHNidP8BACoCAAAAAAFAQg8AAAAAABepFG6Rty1Vk+fUOR4v9E6R6YXDFkHwhwAAAAAAAA==",
```

Comment:
should probably define `seqOfPsbts` and then have `val psbt = seqOfPsbts.head` to make the test less fragile to future changes to the seq, also use `Vector` instead of `Seq` for better performance as a general rule (`Vector` is a subtype of `Seq`)